src/k-provider/ovp/regex-action-handler.js

```diff
@@ -0,0 +1,147 @@
+   * @function _pingECDNAndReplaceUrls
+   * @param {ProviderMediaConfigObject} mediaConfig - The media config
+   * @param {KalturaAccessControlModifyRequestHostRegexAction} regexAction - The regex action
+   * @param {string} urlPing - The url to ping
+   * @returns {Promise<ProviderMediaConfigObject>} - The media config with old or modified urls
+   * @static
+   * @private
+   */
+  static _pingECDNAndReplaceUrls(
+    mediaConfig: ProviderMediaConfigObject,
+    regexAction: KalturaAccessControlModifyRequestHostRegexAction,
+    urlPing: string
+  ): Promise<ProviderMediaConfigObject> {
+    return new Promise(resolve => {
+      const req = new XMLHttpRequest();
+      req.open('GET', urlPing);
+      req.timeout = regexAction.checkAliveTimeoutMs;
+      req.onreadystatechange = () => {
+        if (req.readyState === 4) {
+          if (req.status === 200) {
+            RegexActionHandler._modifyUrls(mediaConfig, regexAction);
+          }
+          resolve(mediaConfig);
+        }
+      };
+      req.ontimeout = () => {
+        RegexActionHandler._logger.warn(`Got timeout while pinging the ECDN url. ping url: ${urlPing}`);
+        resolve(mediaConfig);
+      };
+      req.send();
+    });
+  }
+
+  /**
+   * Handles regex action
+   * @function handleRegexAction
+   * @param {ProviderMediaConfigObject} mediaConfig - The media config
+   * @param {Map<string, Function>} data - The response data
+   * @returns {ProviderMediaConfigObject} - The media config with old or modified urls
+   * @static
+   */
+  static async handleRegexAction(mediaConfig: ProviderMediaConfigObject, data: Map<string, Function>): ProviderMediaConfigObject {
+    let cdnUrl = OVPConfiguration.get()?.cdnUrl || '';
+    const regexAction = RegexActionHandler._extractRegexActionFromData(data);
+    if (regexAction && regexAction.pattern && regexAction.replacement) {
+      const regExp = new RegExp(regexAction.pattern, 'i');
+      if (cdnUrl.match(regExp)) {
+        cdnUrl = cdnUrl.replace(regExp, regexAction.replacement);
+        if (regexAction.checkAliveTimeoutMs && regexAction.checkAliveTimeoutMs > 0) {
+          let urlToPing = cdnUrl + '/api_v3/service/system/action/ping/format/1';
+          const finalMediaConfig = await RegexActionHandler._pingECDNAndReplaceUrls(mediaConfig, regexAction, urlToPing);
+          return finalMediaConfig;
+        } else {
+          RegexActionHandler._modifyUrls(mediaConfig, regexAction);
+          return mediaConfig;
+        }
+      }
+    } else {
+      return mediaConfig;
+    }
+  }
+
+  /**
+   * Modify the urls
+   * @function _modifyUrls
+   * @param {ProviderMediaConfigObject} mediaConfig - The media config
+   * @param {KalturaAccessControlModifyRequestHostRegexAction} regexAction - The regex action
+   * @returns {void}
+   * @static
+   * @private
+   */
+  static _modifyUrls(mediaConfig: ProviderMediaConfigObject, regexAction: KalturaAccessControlModifyRequestHostRegexAction) {
```

Comment:
I would name it in a more informative name such as replaceHost (according to what it does)